clients/command/src/main/java/org/eclipse/hono/client/command/ProtocolAdapterCommandConsumerFactory.java

```diff
@@ -0,0 +1,188 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.command;
+
+import java.time.Duration;
+import java.util.function.Function;
+
+import org.eclipse.hono.util.Lifecycle;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+
+/**
+ * A Protocol Adapter factory for creating consumers for commands targeted at devices.
+ */
+public interface ProtocolAdapterCommandConsumerFactory extends Lifecycle {
+
+    /**
+     * Creates a command consumer for a device.
+     * <p>
+     * For each device only one command consumer may be active at any given time. Invoking this method multiple times
+     * with the same parameters will each time overwrite the previous entry.
+     * <p>
+     * It is the responsibility of the calling code to properly close a consumer
+     * once it is no longer needed by invoking its {@link ProtocolAdapterCommandConsumer#close(boolean, SpanContext)}
+     * method.
+     *
+     * @param tenantId The tenant to consume commands from.
+     * @param deviceId The device for which the consumer will be created.
+     * @param commandHandler The handler to invoke with every command received. The handler must invoke one of the
+     *                       terminal methods of the passed in {@link CommandContext} in order to settle the command
+     *                       message transfer and finish the trace span associated with the {@link CommandContext}.
+     *                       The future returned by the handler indicates the outcome of handling the command.
+     * @param lifespan The time period in which the command consumer shall be active. Using a negative duration or
+     *                 {@code null} here is interpreted as an unlimited life span. The guaranteed granularity
+     *                 taken into account here is seconds.
+     * @param context The currently active OpenTracing span context or {@code null} if no span is currently active.
+     *                An implementation should use this as the parent for any span it creates for tracing
+     *                the execution of this operation.
+     * @return A future indicating the outcome of the operation.
+     *         <p>
+     *         The future will be completed with the newly created consumer once the link
+     *         has been established.
+     *         <p>
+     *         The future will be failed with a {@code org.eclipse.hono.client.ServiceInvocationException} with an
+     *         error code indicating the cause of the failure.
+     * @throws NullPointerException if any of tenant, device ID or command handler is {@code null}.
+     */
+    Future<ProtocolAdapterCommandConsumer> createCommandConsumer(
+            String tenantId,
+            String deviceId,
+            Function<CommandContext, Future<Void>> commandHandler,
+            Duration lifespan,
+            SpanContext context);
```

Comment:
I think it would be better to only have `createCommandConsumer` methods with the `sendEvent` parameter here. This makes it clearer what the behaviour is.

Reply:
agree, the simplification is applied.